rfc/0006-kubewarden-versioning.md

```diff
@@ -0,0 +1,210 @@
+
+It would help users in understanding what version of Kubewarden they
+are about to install or execute.
+
+## Examples / User Stories
+[examples]: #examples
+
+### User story #1
+
+> Users do not need to keep track of component versions and how they
+> work together. They just need to care about running a specific
+> Kubewarden version.
+
+### User story #2
+
+> Users are able to easily understand that a new version of Kubewarden
+> is available, and that more than one component might be upgraded in
+> order to move to the new Kubewarden version.
+
+### User story #3
+
+> A user has an issue with Kubewarden. Given its multiple component
+> nature, we would like to know what was the Kubewarden version that
+> got installed, and through what versions it was upgraded.
+
+### User story #4
+
+> As Kubewarden maintainers we want to define a version skew policy
+> that is easy to be understood and implement.
+> Something similar to what [Kubernetes does](https://kubernetes.io/releases/version-skew-policy/)
+
+# Detailed design
+[design]: #detailed-design
+
+## Many components, one experience
+
+Kubewarden core components, which are exposed to the user are:
+
+- Helm Charts
+- Kubewarden Controller
+- Policy Server
+- kwctl
+
+As a second dimension, there are also policies and their respective
+SDK's. This proposal is going to ignore this dimension on purpose,
+because Kubewarden runtime components (`policy-server` and `kwctl`)
+should be compatible with policies previously built, even with older
+language SDK's.
+
+We want to have a single version for these "public" Kubewarden
+components. This version is going to be expressed using
+[Semantic Versioning](https://semver.org/).
+
+The Kubewarden Controller, the Policy Server and kwctl components are going
+to share the same *Major* and *Minor* version numbers. Each project is
+going to have an independent *Patch* level number.
+
+All the non-visible components of Kubewarden (e.g. the different
+Rust libraries) and the policies are going to retain their
+existing version numbers, release strategy and, generally speaking,
+independence.
+
+### Helm charts
+
+Helm charts have two kind of version numbers:
+
+* `version`: a SemVer 2 version specific to the helm chart
+* `appVersion`: the version of the app that the chart contains
+
+The helm charts will keep their own independence when it comes to
+the `version` attribute.
+On the other hand, the `appVersion` attribute
+will always be set to the `<Major>.<Minor>` version of the
+Kubewarden stack.
+
+> See the official documentation about
+[`Chart.yaml`](https://helm.sh/docs/topics/charts/#the-chartyaml-file)
+for more information.
+
+## Examples
+
+This section provides clarity about how the proposal would work
+using some concrete real-world scenarios.
+
+### A new release Kubewarden takes place
+
+A new version of Kubewarden stack has to be released. That can
+happen because a new important feature/bugfix has been added,
+because of marketing reasons,...
```

Comment:
(opening thread here, loosely related)

I get the marketing reasons, but then, it's not semver I suppose. As a user, the most valuable info for me is if there are backwards-incompatible changes. I want to know if I can blindly upgrade Kubewarden, without needing to redeploy, halt cluster workloads, or make changes.

Normally that would be provided by the helm chart, but the helm chart would get its `version` bumped a major version regardless.